### PR TITLE
Drop legacy results api

### PR DESCRIPTION
**Pull request by arhimondr**
## Description

In favor of `/async` version. Point the original endpoint to /v1/task/async.

## Motivation and Context

Async getResults API has been enabled by default

## Test Plan

CI

## Contributor checklist

- [X] Please make sure your submission complies with our [development](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#development), [formatting](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#formatting), [commit message](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#commit-formatting-and-pull-requests), and [attribution guidelines](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#attribution).
- [X] PR description addresses the issue accurately and concisely.  If the change is non-trivial, a GitHub Issue is referenced.
- [X] Documented new properties (with its default value), SQL syntax, functions, or other functionality.
- [X] If release notes are required, they follow the [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines).
- [X] Adequate tests were added if applicable.
- [X] CI passed.

## Release Notes

```
== RELEASE NOTES ==

General Changes
* Deprecate exchange.async-page-transport-enabled configuration property
```

### CODE REVIEW

**Comment by arhimondr** [comment]
Still WIP

**Comment by arhimondr** [comment]
Ready for review

File: presto-main/src/main/java/com/facebook/presto/server/TaskResource.java [hunk 1]
```diff
@@ -298,65 +275,6 @@ public TaskInfo deleteTask(
         return taskInfo;
     }
 
-    @GET
-    @Path("{taskId}/results/{bufferId}/{token}")
-    @Produces(PRESTO_PAGES)
-    public void getResults(
```
Comment (spershin) [inline presto-main/src/main/java/com/facebook/presto/server/TaskResource.java:304]: @arhimondr 
Why is this going way?

Comment (arhimondr) [reply presto-main/src/main/java/com/facebook/presto/server/TaskResource.java:304]: It's an old implementation based on blocking IO. [AsyncPageTransportServlet.java](https://github.com/prestodb/presto/pull/21772/files#diff-a338733a9579ecb50bd4a1f6d508bb66ad7c3e8888c7165feda720cdc60095c9) implements in in a non-blocking way. Semantic wise both are identical.

File: presto-main/src/main/java/com/facebook/presto/server/AsyncPageTransportForwardFilter.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+import java.io.IOException;
+import java.util.regex.Pattern;
+
+/**
+ *  It is not possible to map the {@link AsyncPageTransportServlet} to "/v1/task/<taskid>/results/*" directly.
+ *  Servlet pattern matching doesn't allow arbitrary globbing (e.g.: /v1/task/*\/results/*).
+ *  Only prefix or suffix matching is allowed (e.g.: /v1/task/*, *\/suffix).
+ *  Hence a more nuanced matching and a forward is required.
+ */
+public class AsyncPageTransportForwardFilter
```
Comment (tdcmeehan) [inline presto-main/src/main/java/com/facebook/presto/server/AsyncPageTransportForwardFilter.java:33]: Why not just implement the rename in one shot, as opposed to forwarding `/v1/task` to `/v1/task/async`?  i.e. why not just use `/v1/task` and remove this filter?

Comment (arhimondr) [reply presto-main/src/main/java/com/facebook/presto/server/AsyncPageTransportForwardFilter.java:33]: Then all the requests starting with `/v1/task` will be routed to the custom servlet, while requests other than `/v1/task*/results/*` have to be routed to `TaskResource`

Comment (tdcmeehan) [reply presto-main/src/main/java/com/facebook/presto/server/AsyncPageTransportForwardFilter.java:33]: I can't think of a better way of doing it.

**Review by tdcmeehan — APPROVED** [review]
LGTM

**Pull request merged**